cpp/src/io/json/reader_impl.cu

```diff
@@ -334,50 +298,54 @@ rmm::device_uvector<uint64_t> reader::impl::find_record_starts(rmm::cuda_stream_
+  size_t end_offset = h_data.size();
 
   // Trim lines that are outside range
-  if (byte_range_size_ != 0 || byte_range_offset_ != 0) {
-    auto h_rec_starts = cudf::detail::make_std_vector_sync(rec_starts, stream);
-
-    if (byte_range_size_ != 0) {
-      auto it = h_rec_starts.end() - 1;
-      while (it >= h_rec_starts.begin() && *it > byte_range_size_) {
-        end_offset = *it;
-        --it;
-      }
-      h_rec_starts.erase(it + 1, h_rec_starts.end());
-    }
+  auto h_rec_starts = cudf::detail::make_std_vector_sync(rec_starts, stream);
 
-    // Resize to exclude rows outside of the range
-    // Adjust row start positions to account for the data subcopy
-    start_offset = h_rec_starts.front();
-    rec_starts.resize(h_rec_starts.size(), stream);
-    thrust::transform(rmm::exec_policy(stream),
-                      rec_starts.begin(),
-                      rec_starts.end(),
-                      thrust::make_constant_iterator(start_offset),
-                      rec_starts.begin(),
-                      thrust::minus<uint64_t>());
+  if (reader_opts.get_byte_range_size() != 0) {
+    auto it = h_rec_starts.end() - 1;
+    while (it >= h_rec_starts.begin() && *it > reader_opts.get_byte_range_size()) {
+      end_offset = *it;
+      --it;
+    }
+    h_rec_starts.erase(it + 1, h_rec_starts.end());
   }
 
+  // Resize to exclude rows outside of the range
+  // Adjust row start positions to account for the data subcopy
+  size_t start_offset = h_rec_starts.front();
+  rec_starts.resize(h_rec_starts.size(), stream);
+  thrust::transform(rmm::exec_policy(stream),
+                    rec_starts.begin(),
+                    rec_starts.end(),
+                    thrust::make_constant_iterator(start_offset),
+                    rec_starts.begin(),
+                    thrust::minus<uint64_t>());
+
   const size_t bytes_to_upload = end_offset - start_offset;
-  CUDF_EXPECTS(bytes_to_upload <= uncomp_size_,
+  CUDF_EXPECTS(bytes_to_upload <= h_data.size(),
                "Error finding the record within the specified byte range.\n");
 
   // Upload the raw data that is within the rows of interest
-  data_ = rmm::device_buffer(uncomp_data_ + start_offset, bytes_to_upload, stream);
+  return cudf::detail::make_device_uvector_async(h_data.subspan(start_offset, bytes_to_upload),
+                                                 stream);
 }
 
-void reader::impl::set_column_names(device_span<uint64_t const> rec_starts,
-                                    rmm::cuda_stream_view stream)
+std::pair<std::vector<std::string>, col_map_ptr_type> get_column_names_and_map(
+  parse_options_view const& parse_opts,
+  host_span<char const> h_data,
+  device_span<uint64_t const> rec_starts,
+  device_span<char const> d_data,
+  rmm::cuda_stream_view stream)
 {
   // If file only contains one row, use the file size for the row size
-  uint64_t first_row_len = data_.size() / sizeof(char);
+  uint64_t first_row_len = d_data.size() / sizeof(char);
```

Comment:
` d_data.size() / sizeof(char);` is curious. Isn't `d_data.size()` counted in bytes? Isn't `sizeof(char) == 1`? (Pardon my ignorance.)

Reply:
afaik, yes.